src/Microsoft.DotNet.Wpf/src/PresentationFramework/Microsoft/Win32/CommonItemDialog.cs

```diff
@@ -2,31 +2,20 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-//
-// 
-// Description:
-//              CommonItemDialog is an abstract class derived from CommonDialog
-//              that implements shared functionality common to all IFileDialog
-//              variants. It provides a common options storage and events handling.
-//
+using MS.Internal.AppModel;
+using MS.Internal.Interop;
+using MS.Win32;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Windows;
+
+// There are THREE definitions of HRESULT. Two in ErrorCodes, and one in wgx_render.cs.
+// wgx_render.cs wins if we don't explicitly define it here.
+using HRESULT = Standard.HRESULT;
```

Comment:
The HRESULT implementation in both the namespaces are a bit different. There are some additional HRESULT members like DISP_E_MEMBERNOTFOUND, DISP_E_TYPEMISMATCH and others in `MS.Internal.Interop`. Additionally, the `make` method in `MS.Internal.Interop` had a couple of asserts which were probably added to cater specific checks. This seems like it can break something. Thoughts?

Similar changes in 
- src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Shell/JumpList.cs
- src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Window.cs

Reply:
If I got it wrong exceptions would potentially change type. That would be the worst of it.

The most notable ones are `WINCODEC_ERR_*` and `WGXERR_*` code handling. My recommendation would be always check and convert these. I think it is probably extremely unlikely people are dependent on `COMException` ever being thrown for these ones.

What ultimately needs to happen is we need to move to the HRESULT struct coming from CsWin32 and write an ConvertHRToException extension for it that maps the WIC and WGX errors to a friendlier exception. I'll try to create a PR that does this.

Reply:
Since we keep suggesting to people that catch-all is bad and specific exceptions should be handled, I would be wary of assumptions on how unlikely this is.

> write an ConvertHRToException

Doesn't `Marshal.ThrowExceptionForHR` not do the right job?

Reply:
@miloush `Marshal.ThrowExceptionForHR` doesn't have any concept of WIC or WGX exceptions. It has some logic for unwrapping exceptions that came from the .NET runtime and mapping for some common Win32 errors and `HRESULT`s. WPF already has the helper that maps WIC and WGX `HRESULT`s to friendlier exceptions, the change I'll do is just make this work with the CsWin32 `HRESULT` definition and always use the one helper that understands these.

Reply:
I've added the WIC `HRESULT`s to System.Private.Windows.Core and will use them to follow up on this topic. https://github.com/dotnet/winforms/pull/12579